posts/2023-03-31-embrace-kotlin.adoc

```diff
@@ -0,0 +1,273 @@
+= Embrace Kotlin: Where to Get Started for Scala Developers
+AnastasiiaL
+v1.0, 2023-03-31
+:title: Embrace Kotlin: Tips & Tricks for Scala Developers to Get Started
+:imagesdir: ../media/2023-03-31-embrace-kotlin
+:lang: en
+:tags: [scala, kotlin]
+
+Recently I have been intrigued by the buzz surrounding Kotlin, another powerful language on the JVM.
+The growing popularity and claims of a seamless transition from Scala to Kotlin have piqued my curiosity,
+prompting me to dive in and explore the language myself.
+In this blog post, I'll share my journey of learning Kotlin from a Scala developer's perspective and provide
+insights into the similarities and differences between the two languages that make the switch feel so smooth.
+
+== Language similarities
+
+Both Kotlin and Scala are statically-typed languages that run on the Java Virtual Machine. They share many features, such as:
+- Support for both functional programming and OO
+- Interoperability with Java
+- Type inference
+- Null safety
+- Extension functions
+- Pattern matching (although more limited in Kotlin)
+These similarities create a strong foundation for Scala developers to quickly grasp Kotlin's syntax and paradigms.
+For me personally it was quite easy to create my first kotlin app without looking up "how-to"s, because structurally there would be nothing new.
+
+== Key differences between Kotlin and Scala:
+
+Despite their similarities, there are some differences between Kotlin and Scala. Notable differences include:
+- Kotlin's syntax is generally more concise and closer to Java than Scala's.
+- Scala has more advanced features, such as support of typeclasses, macros, more powerful and extensive pattern matching.
+- Java Interoperability: Kotlin prioritizes seamless Java interoperability, allowing developers to use Kotlin code alongside
+Java code without any issues. Scala has Java interoperability as well, but due to its more complex type system and language features, some constructs might require additional effort to work seamlessly with Java code.
+- Ecosystem and Community: Both Kotlin and Scala have active and growing caommunities.
+However, Kotlin's community is rapidly expanding, particularly in the Android development space, while Scala's community is more focused on big data, data processing, and backend services.
+
+== Understanding Kotlin's syntax
+
+While Kotlin's syntax is inspired by Scala, it's a bit more concise but also Java-like.
+You will quickly get used to the new syntax as it is very similar - I'll show some examples.
+
+- Defining functions:
```

Comment:
Wouldn't it look nicer if you converted these list items into level 3 headers? (same for _Immutable values_, _Mutable values_, _Data classes_, etc.)